### PR TITLE
Codegen: preserve comments

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Preserve and take over class/method boilerplate comments from input `TW*.h` files to generated source files.

## How to test

Check generated Swift/Java sources, check that comments are shown for methods in Xcode/AndroidStudio.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Comment by catenocrypt** [comment]
As first step method comments in Swift are included, generated file looks like this:

image

**Comment by catenocrypt** [comment]
Comments are preserved into all entities (class, struct, enum, and their methods & properties), on Swift side. To do in Kotlin templates as well.

File: codegen/lib/templates/java/header.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2017-2021 Trust Wallet.
+// Copyright © 2017-2022 Trust Wallet.
```
Comment (hewigovens) [inline codegen/lib/templates/java/header.erb:1]: I think we can get current year from ruby code in a separate pr

File: codegen/test/test_parser.rb [hunk 2]
```diff
@@ -44,22 +44,26 @@ def test_parse_invalid_method
...
   def test_parse_method_discardable_result
     parser = Parser.new(path: '', string: '
+      // This is a sample file
       TW_EXTERN_C_BEGIN
       struct TWEthereumAbiFunction;
 
+      // Ethereuem ABI helpers
       TW_EXPORT_CLASS
       struct TWEthereumAbiEncoder;
-      /// Encode function to Eth ABI binary
```
Comment (hewigovens) [inline codegen/test/test_parser.rb:52]: should be `///`?

**Review by hewigovens — APPROVED** [review]
Good work, left 2 comments

File: codegen/test/test_parser.rb (hunk 2 above)
Comment (catenocrypt) [reply codegen/test/test_parser.rb:52]: ?

File: codegen/lib/templates/java/header.erb (hunk 1 above)
Comment (catenocrypt) [reply codegen/lib/templates/java/header.erb:1]: Good idea

File: codegen/test/test_parser.rb (hunk 2 above)
Comment (catenocrypt) [reply codegen/test/test_parser.rb:52]: Yes, all lines starting with `//` (2 slashes) before an entity is being copied over to the generated source file as well.  So yes, this includes lines starting with 3 (or more) slashes as well.

**Pull request merged**